### PR TITLE
Set initial state for profile and newsFeed reducers.

### PR DESCRIPTION
**Pull request by stefanv**
This avoids checks inside components to ensure that state is defined and
of correct form.

It would be good to review other components that do this kind of checking too.

---

The major changes here are now:

- The backend never returns undefined for profile preferences, but rather an empty dictionary.
- The newsfeed component now checks for its preferences, and if not found applies a set of default preferences.
  - TODO: The default preferences should probably be merged with whatever is provided by the user, so we can ensure that the widget always has all of its configuration parameters.
- Random cleanups, but specifically API.js, which I rewrote to give more logical error responses: a) connection error b) JSON decoding error c) actual 400 and similar errors that are reported by the backend.

### CODE REVIEW

**Comment by acrellin** [comment]
@stefanv the issue _wasn't_ that the default value of `profile.preferences` was an empty dict and didn't have a `newsFeed` attribute, which is why there was still an error when I removed all of the attribute checks _except_ for `Object.prototype.hasOwnProperty.call(profile.preferences, "newsFeed")` -- even with that being checked, the error was occurring. This PR doesn't address the issue -- it wasn't that the default redux state wasn't properly initialized. I don't know that we even want to bake all of the possible component config state into the redux state...

**Comment by stefanv** [comment]
What was the issue that you witnessed then?  I can reproduce these failures locally, and am trying to see where things go wrong.  I see that the reducers are initialized correctly.

**Comment by acrellin** [comment]
Can you reproduce them locally even with the new default reducer values?

**Comment by stefanv** [comment]
Not via the browser, but via the tests.  Digging a bit deeper into baselayer now :)

File: static/js/components/GroupList.jsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import SourceList from './SourceList';
 
 const GroupList = ({ title, listSources, groups }) => {
   if (!groups || groups.length === 0) {
-    groups = useSelector((state) => state.groups.latest);
+    groups = useSelector((state) => state.groups.user);
```
Comment (acrellin) [inline static/js/components/GroupList.jsx]: Nice, I've been thinking we should rename this for a long time!

File: static/js/ducks/profile.js [hunk 2]
```diff
@@ -52,7 +52,9 @@ const initialState = {
   roles: [],
   acls: [],
   tokens: [],
-  preferences: {}
+  preferences: {
+    newsFeed: {}
```
Comment (acrellin) [inline static/js/ducks/profile.js]: This seems unnecessary, and I'm not sure we want to have to keep updating this default value every time we add a new configurable component/widget. The `NewsFeed` component already checks whether `preferences` has  a `newsFeed` attr, and supplies a default value if not. I think it'd simplify the process of adding new features to just have `preferences` default to `{}` here.

Comment (stefanv) [reply static/js/ducks/profile.js]: Removed, let's see how the tests go!

**Comment by acrellin** [comment]
Looks great! Thanks @stefanv! In it goes :rocket:

**Pull request merged**